backend/tests/unit/danswer/llm/answering/stream_processing/test_quote_processing.py

```diff
@@ -278,3 +281,30 @@ def test_json_answer_split_tokens() -> None:
     )
 
     assert expected_answer == actual
+
+
+def test_process_model_tokens_quotes() -> None:
+    tokens = [
+        "```",
+        "json",
+        "\n",
+        "{",
+        '"answer": "This is a simple ',
+        "answer.",
+        '",\n"',
+        'quotes": ["Document"]',
+        "\n}",
+        "\n",
+        "```",
+    ]
+
+    gen = process_model_tokens(tokens=iter(tokens), context_docs=mock_docs)
+
+    actual_count = 0
```

Comment:
can we check that the answer is correct here as well? This way we catch a case where having a valid quote causes the answer to be incorrect (due to some bug).